### PR TITLE
frontend/feature: #382 generate session token

### PR DESCRIPTION
**Pull request by chakri642**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by Vignesh-772** [comment]
resolve conflicts

File: Frontend/ui-customer/src/Helpers/Storage.purs [hunk 1]
```diff
@@ -59,6 +59,7 @@ data KeyStore
   | FINDING_QUOTES_START_TIME
   | FINDING_QUOTES_POLLING
   | FLOW_STATUS_DATA
+  | UNIQUE_ID
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Helpers/Storage.purs:62]: rename it to `SESSION_ID`

File: Frontend/ui-customer/src/Flow.purs [hunk 2]
```diff
@@ -425,6 +425,7 @@ homeScreenFlow = do
         modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{currentStage = SearchLocationModel ,rideRequestFlow = false, isSearchLocation = SearchLocation, isSrcServiceable = false, isSource = Just true, isRideServiceable = false}})
         homeScreenFlow
         else pure unit
+      _ <- pure $ setValueToLocalStore UNIQUE_ID (generateUId unit)
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Flow.purs:428]: I think we are doing the `/rideSearch` api in two places in Flow file update the `session_id` in all places

File: Frontend/ui-customer/src/Helpers/Utils.js [hunk 3]
```diff
@@ -555,4 +555,18 @@ exports["storeOnResumeCallback"] = function (cb) {
       }
     }
   }
+}
+
+exports["generateUId"] = function () {
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Helpers/Utils.js:560]: rename the function name to meaning full name

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

**Comment by Vignesh-772** [comment]
I can be merged once the backend changes for this are done

File: Frontend/ui-driver/src/Services/Backend.purs [hunk 4]
```diff
@@ -23,7 +23,7 @@ import Presto.Core.Types.API (Header(..), Headers(..))
 import Presto.Core.Types.Language.Flow (Flow, callAPI, doAff)
 import Helpers.Utils (decodeErrorCode, decodeErrorMessage, toString,getTime)
 import Foreign.Generic (encode)
-import JBridge (setKeyInSharedPrefKeys,toast,factoryResetApp, toggleLoader, stopLocationPollingAPI, Locations, getVersionName)
+import JBridge (setKeyInSharedPrefKeys,toast,factoryResetApp, toggleLoader, stopLocationPollingAPI, Locations, getVersionName, generateSessionId)
```
Comment (Vignesh-772) [inline Frontend/ui-driver/src/Services/Backend.purs:26]: remove unused import

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

**Review by Vignesh-772 — APPROVED** [review]

File: Frontend/ui-customer/src/Flow.purs [hunk 5]
```diff
@@ -87,6 +87,10 @@ baseAppFlow gPayload = do
   _ <- pure $ setValueToLocalStore TEST_POLLING_INTERVAL "1500.0"
   _ <- pure $ setValueToLocalStore TEST_POLLING_COUNT "113"
   _ <- pure $ setValueToLocalStore RATING_SKIPPED "false"
+  if ((getValueToLocalStore SESSION_ID == "__failed") || (getValueToLocalStore SESSION_ID == "(null)")) then do
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Flow.purs:90]: if you want you can do something like 
`when (condition) $ do
      setValueToLocalStore SESSION_ID (generateSessionId unit)`

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 6]
```diff
@@ -131,6 +131,7 @@ screen initialState =
                   launchAff_ $ flowRunner $ getQuotesPolling (getValueToLocalStore TRACKING_ID) GetQuotesList Restart pollingCount (fromMaybe 0.0 (NUM.fromString (getValueToLocalStore TEST_POLLING_INTERVAL))) push initialState
               ConfirmingRide -> launchAff_ $ flowRunner $ confirmRide GetRideConfirmation 5 1000.0 push initialState
               HomeScreen -> do
+                _ <- pure $ setValueToLocalStore SESSION_ID (generateSessionId unit)
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:134]: do we want same session id for a single ride ? if yes then can we call it where we are searching for ride

File: Frontend/ui-customer/src/Flow.purs (hunk 5 above)
[the remark this answers is not in the document]
Comment (chakri642) [reply Frontend/ui-customer/src/Flow.purs:90]: It is for IOS checking whether the value is in local storage or not

[the remark this answers is not in the document]
Comment (ashishsingh101) [reply Frontend/ui-customer/src/Flow.purs:90]: okay

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**